x-pack/solutions/observability/plugins/streams/server/routes/streams/edit.ts

```diff
@@ -207,3 +215,35 @@ async function validateStreamChildren(
     }
   }
 }
+
+/*
+ * Changes to mappings (fields) and processing rules are not allowed on the root stream.
+ * Changes to routing rules are allowed.
+ */
+async function validateRootStreamChanges(
+  scopedClusterClient: IScopedClusterClient,
+  nextStreamDefinition: WiredStreamDefinition
+) {
+  const oldDefinition = (await readStream({
```

Comment:
can we move this out of this function and into the route handler? IMHO it's really easy to start duplicating requests in multi-step processes like these because getting a dependency becomes an implementation detail.